### PR TITLE
Feat/5111  merge header and sidebar

### PR DESCRIPTION
**Pull request by gbonaspetti**
#5111 

## Checkpoints

**For developers**

- [x] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [x] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [x] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)
- [x] Make sure that:
  - if there are modifications in the Tracim configuration files (eg. `development.ini`), they are documented in `backend/doc/setting.md`
  - any migration process required for existing instances is documented
  - relevant people for these changes are notified
- [x] Original authors of the features included in a multi-feature branch (maintenance fixes -> develop, security fixes -> develop, …) should be part of the reviewers, especially if you encountered merge conflicts.

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [x] Manual, quality tests have been done

### CODE REVIEW

File: frontend/src/container/Sidebar.jsx [hunk 1]
```diff
@@ -37,23 +43,20 @@ export class Sidebar extends React.Component {
...
       foldedSpaceList: [],
-      sidebarClose: isMobile
+      isSidebarClose: isMobile,
+      showSpaceList: true,
+      showUserItems: true
     }
 
     props.registerCustomEventHandlerList([
-      { name: CUSTOM_EVENT.SHOW_CREATE_WORKSPACE_POPUP, handler: this.handleShowCreateWorkspacePopup }
+      { name: CUSTOM_EVENT.SHOW_CREATE_WORKSPACE_POPUP, handler: this.handleClickNewWorkspace }
```
Comment (Haskaris) [inline frontend/src/container/Sidebar.jsx:52]:
```suggestion
      { name: CUSTOM_EVENT.SHOW_CREATE_WORKSPACE_POPUP, handler: this.handleClickNewSpace }
```
I'd suggest to rename the handleClickNewWorkspace function to handleClickNewSpace

File: frontend/src/container/Sidebar.jsx [hunk 2]
```diff
@@ -37,23 +43,20 @@ export class Sidebar extends React.Component {
     this.state = {
       activeWorkspaceId: NO_ACTIVE_SPACE_ID,
       foldedSpaceList: [],
-      sidebarClose: isMobile
+      isSidebarClose: isMobile,
```
Comment (Haskaris) [inline frontend/src/container/Sidebar.jsx:46]:
```suggestion
      isSidebarClosed: isMobile,
```

File: frontend_app_thread/src/css/index.styl [hunk 3]
```diff
@@ -43,6 +43,6 @@
 
 @media (max-width: max-xs)
   .thread .timeline
-    height "calc(100vh - %s)" % (headerHeight * 2 + 25px) // INFO - 2021-08-08 - GB - 25px is the thread__contentpage__top height
+    height "calc(100vh - %s)" % (headerHeight + 25px) // INFO - 2021-08-08 - GB - 25px is the thread__contentpage__top height
```
Comment (Haskaris) [inline frontend_app_thread/src/css/index.styl:46]: Do we keep the header in mobile?

**Review by Haskaris — CHANGES_REQUESTED** [review]
Maybe check if the height 100% is required everywhere. It seems that we used it because of the calculation to remove the header height

File: frontend_app_thread/src/css/index.styl (hunk 3 above)
Comment (gbonaspetti) [reply frontend_app_thread/src/css/index.styl:46]: No, I didn't make a lot of tests on mobile yet, but I think the 2 headerHeights were: main header + app header. So I deleted only one.

**Review by Haskaris — APPROVED** [review]

**Comment by Haskaris** [comment]
Tested on branch.

**Pull request merged**